jose-jwt/JWTSettings.cs

```diff
@@ -9,6 +10,28 @@ namespace Jose
     /// </summary>
     public class JwtSettings
     {
+        public JwtSettings()
+        {
+            // By giving the Unix ECDHKeyManagement implementation to windows, we enable windows version of it to work with not only CngKey but also ECDiffieHellman.
+            // Initially this was implemented separately, but unit tests were failing on windows due to the lack of ECDiffieHellman support. 
+            // Since we don't know what the keys will be provided until runtime, and the registration happens before runtime, we need to make sure 
+            // on windows it will also supports ECDiffieHellman. 
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                keyAlgorithms.Add(JweAlgorithm.ECDH_ES, new EcdhKeyManagementWin(true, new EcdhKeyManagementUnix(true)));
+                keyAlgorithms.Add(JweAlgorithm.ECDH_ES_A128KW, new EcdhKeyManagementWinWithAesKeyWrap(128, new AesKeyWrapManagement(128), new EcdhKeyManagementUnixWithAesKeyWrap(128, new AesKeyWrapManagement(128))));
+                keyAlgorithms.Add(JweAlgorithm.ECDH_ES_A192KW, new EcdhKeyManagementWinWithAesKeyWrap(192, new AesKeyWrapManagement(192), new EcdhKeyManagementUnixWithAesKeyWrap(192, new AesKeyWrapManagement(192))));
+                keyAlgorithms.Add(JweAlgorithm.ECDH_ES_A256KW, new EcdhKeyManagementWinWithAesKeyWrap(256, new AesKeyWrapManagement(256), new EcdhKeyManagementUnixWithAesKeyWrap(256, new AesKeyWrapManagement(256))));
+            }
+            else

```

Comment:
Are we sure we want blind `else` here? Or be more explicit?

Reply:
What would you suggest to be the explicitness of the `else`? My reasoning for it was that only windows supports CngKey, thus anything that's not Windows would be the Unix version of the EcdhKeyManagement.

Reply:
Well, 2 things basically. OSPlatform defines 4 types: windows, linux, freebsd, osx

1. Are we sure it works on freebsd/osx same way as on linux?
2. When .NET introduces fifth value in to enumeration (whatever it can be) will library still work with newly given OS? And more importantly will it open any potential security issue?

Just trying to be more explicit in what we know really works vs. others.